crossbeam-skiplist/src/base.rs

```diff
@@ -882,6 +893,9 @@ where
                 // The reference count is initially two to account for:
                 // 1. The entry that will be returned.
                 // 2. The link at the level 0 of the tower.
+
+                // create value before creating node, so extra allocation doesn't happen if value() function panics
+                let value = value();
```

Comment:
Could you move these two lines to before `let height = self.random_height();` and its comment?

Reply:
Taken care.